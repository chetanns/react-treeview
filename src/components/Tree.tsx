import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import TreeView from "@material-ui/lab/TreeView";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";
import TreeItem from "@material-ui/lab/TreeItem";
import "../utils/string.extensions";
import dataJson from "../resources/app-data.json";
interface RenderTree {
  id: string;
  name: string;
  children?: RenderTree[];
}

const data: RenderTree = {
  id: "root",
  name: "Parent",
  children: [
    {
      id: "1",
      name: "Child - 1",
    },
    {
      id: "3",
      name: "Child - 3",
      children: [
        {
          id: "4",
          name: "Child - 4",
        },
      ],
    },
  ],
};

const useStyles = makeStyles({
  root: {
    height: 110,
    flexGrow: 1,
    maxWidth: 400,
  },
});

export default function RecursiveTreeView() {
  const classes = useStyles();

  console.log(`Testing -- > ${dataJson[0].appGroup}`);

  const map = new Map<string, string>();

  map.set("1", "test-1");
  map.set("2", "test-2");
  map.set("3", "test-3");
  map.set("4", "test-4");

  console.log("testing".getByDefault());
  console.log(map.getByDefault("5", "Default-value"));

  const renderTree = (nodes: RenderTree) => (
    <TreeItem key={nodes.id} nodeId={nodes.id} label={nodes.name}>
      {Array.isArray(nodes.children)
        ? nodes.children.map((node) => renderTree(node))
        : null}
    </TreeItem>
  );

  // const handleChange = (event: any, node:any)=>{
  //     console.log("Testing---"+ node.name);
  //   //console.log(node.name+"-"+ Array.isArray(node))
  // }

  return (
    <TreeView
      className={classes.root}
      defaultCollapseIcon={<ExpandMoreIcon />}
      defaultExpanded={["root"]}
      defaultExpandIcon={<ChevronRightIcon />}
    >
      {renderTree(data)}
    </TreeView>
  );
}
