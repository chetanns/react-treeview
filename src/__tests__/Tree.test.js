import React from 'react';
import {render,fireEvent} from '@testing-library/react'

import Tree from '../components/Tree';

describe('Tree',()=>{
    test('render tree component',()=>{
        const {getByText} = render(<Tree/>);

        const parent = getByText('Parent')

        expect(parent).toBeInTheDocument();

        fireEvent.click(parent);

        const child = getByText('Child - 1');

        expect(child).toBeInTheDocument();
    })
})
