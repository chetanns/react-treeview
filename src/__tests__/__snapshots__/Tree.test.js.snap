// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree should render correctly in debug mode 1`] = `
<WithStyles(ForwardRef(TreeView))
  className="makeStyles-root-1"
  defaultCollapseIcon={<Memo />}
  defaultExpandIcon={<Memo />}
  defaultExpanded={
    Array [
      "root",
    ]
  }
>
  <WithStyles(ForwardRef(TreeItem))
    key="root"
    label="Parent"
    nodeId="root"
  >
    <WithStyles(ForwardRef(TreeItem))
      key="1"
      label="Child - 1"
      nodeId="1"
    />
    <WithStyles(ForwardRef(TreeItem))
      key="3"
      label="Child - 3"
      nodeId="3"
    >
      <WithStyles(ForwardRef(TreeItem))
        key="4"
        label="Child - 4"
        nodeId="4"
      />
    </WithStyles(ForwardRef(TreeItem))>
  </WithStyles(ForwardRef(TreeItem))>
</WithStyles(ForwardRef(TreeView))>
`;
